chalice/deploy/packager.py

```diff
@@ -1,60 +1,72 @@
 from __future__ import print_function
+import sys
 import hashlib
 import inspect
 import os
 import shutil
 import subprocess
 import zipfile
+import tempfile
+import contextlib
+import pip
 
 import virtualenv
-from typing import Any, Optional  # noqa
+from typing import Any, List, Optional, Tuple, Iterable, Callable  # noqa
+from chalice.compat import StringIO
+from chalice.utils import OSUtils
 
 import chalice
 from chalice import compat, app
 
 
+class InvalidSourceDistributionNameError(Exception):
+    pass
+
+
 class LambdaDeploymentPackager(object):
     _CHALICE_LIB_DIR = 'chalicelib'
     _VENDOR_DIR = 'vendor'
 
-    def _create_virtualenv(self, venv_dir):
-        # type: (str) -> None
-        virtualenv.create_environment(venv_dir)
+    def __init__(self, osutils=None):
+        # type: (Optional[OSUtils]) -> None
+        if osutils is None:
+            osutils = OSUtils()
+        self._osutils = osutils
 
-    def create_deployment_package(self, project_dir, package_filename=None):
+    def _get_requirements_file(self, project_dir):
+        # type: (str) -> str
+        # Gets the path to a requirements.txt file out of a project dir path
+        return os.path.join(project_dir, 'requirements.txt')
```

Comment:
We probably want to include this functionality in the OSUtils. In general any call to the `os` library should probably live in the `OSUtils` class.